test/test_response.py

```diff
@@ -560,6 +560,38 @@ def test_io_closed_consistently(self, sock: socket.socket) -> None:
         finally:
             hlr.close()
 
+    @pytest.mark.parametrize("read_amt", (None, 3))
```

Comment:
Can we add a case too for when `read_amt` is less than the total length (thus requiring more than one `read1` call?) and have assertions on the in-between state and final state.

Reply:
I added two more cases: one suggested by you and one reading 0 bytes